### PR TITLE
[TE] frontend - aaronucsd/Add new tab components

### PR DESCRIPTION
**Pull request by aaronucsd**
screen shot 2018-04-27 at 10 20 05 am
screen shot 2018-04-30 at 1 32 17 pm

1. Added new common-tabs components (and its sub components - tablist, tabpanel, tab)
2. Added new styles for this common-tabs component
3. Added integration tests for the new component
4. Updated RCA page to use this new component
5. Documented  example of usage (see below).
```
  @example
    {{#shared/common-tabs selection=activeTab activeTab=activeTab as |tabs|}}
      {{#tabs.tablist as |tablist|}}
        {{#tablist.tab name="metrics"}}Metrics{{/tablist.tab}}
        {{#tablist.tab name="dimensions"}}Dimensions{{/tablist.tab}}
      {{/tabs.tablist}}

      {{!-- metrics --}}
      {{#tabs.tabpanel name="metrics"}}
        {{!-- IMPORTANT metrics CODE --}}
      {{/tabs.tabpanel}}
      {{!-- dimensions --}}
      {{#tabs.tabpanel name="dimensions"}}
        {{!-- IMPORTANT dimensions CODE --}}
      {{/tabs.tabpanel}}
    {{/shared/common-tabs}}
```

### CODE REVIEW

File: thirdeye/thirdeye-frontend/tests/integration/pods/components/shared/common-tabs/component-test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { module, test } from 'qunit';
```
Comment (newsummit) [inline thirdeye/thirdeye-frontend/tests/integration/pods/components/shared/common-tabs/component-test.js]: So for testing interaction of tabs and tabpanels, I assume RCA acceptance test will handle that, correct?

File: thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs [hunk 2]
```diff
@@ -152,140 +152,122 @@
...
-              Events
-            </a>
-          </div>
-          <div class="card-container__subnav">
-            <a class="thirdeye-link thirdeye-link--nav {{if (eq activeTab "trend") "thirdeye-link--active"}}" {{action (mut activeTab) "trend"}}>
-              Trend
-            </a>
-          </div>
-        </div>
+        {{#shared/common-tabs selection=activeTab activeTab=activeTab as |tabs|}}
```
Comment (newsummit) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs]: At first look, it seems the tabs component set will be able to handle "sub-tabs" in any given `tabpanel`. Any thoughts on that?

File: thirdeye/thirdeye-frontend/app/pods/components/shared/common-tabs/tablist/component.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/**
+ * @description Tablist Component
+ * @summary Displays a tablist body. This is meant for usage with `Common-tabs` component.
+ * @module components/shared/common-tabs/tablist
+ * @example
```
Comment (newsummit) [inline thirdeye/thirdeye-frontend/app/pods/components/shared/common-tabs/tablist/component.js]: Thanks for providing the example. Am I seeing double `@example`?

**Comment by justYves** [comment]
nitpick: 
- in the provided screenshot, looks like there's a double border (around the tab card), background-- color seems a bit off as well

**Comment by aaronucsd** [comment]
"nitpick:

in the provided screenshot, looks like there's a double border (around the tab card),
background color seems a bit off as well"

@justYves Yes t was intended. Have discussed with Kishore on these changes. The idea is to help make the tab more dominate and noticeable as customers reports not knowing those are tabs.

File: thirdeye/thirdeye-frontend/tests/integration/pods/components/shared/common-tabs/component-test.js (hunk 1 above)
Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/tests/integration/pods/components/shared/common-tabs/component-test.js]: Yes. Existing has been updated.

**Pull request merged**